airbyte-ci/connectors/regression-testing/src/regression_testing/main.py

```diff
@@ -0,0 +1,191 @@
+    config_path: Optional[str],
+    catalog_path: Optional[str],
+    state_path: Optional[str],
+):
+    if control_image_name in ("dev", "latest"):
+        control_image_name = f"airbyte/{connector_name}:{control_image_name}"
+
+    if target_image_name in ("dev", "latest"):
+        target_image_name = f"airbyte/{connector_name}:{target_image_name}"
+
+    # TODO: add backend options
+    if not output_directory:
+        raise NotImplementedError(f"An output directory is required; only file backends are currently supported.")
+
+    config = get_connector_config(config_path)
+    catalog = ConfiguredAirbyteCatalog.parse_file(catalog_path) if catalog_path else None
+    state = get_state(state_path) if state_path else None
+
+    async with (dagger.Connection(config=dagger.Config(log_output=sys.stderr)) as client):
+        control_connector = await get_connector(client, connector_name, control_image_name)
+        target_connector = await get_connector(client, connector_name, target_image_name)
+
+        # TODO: maybe use proxy to cache the response from the first round and use the cache for the second round
+        #   (this may only make sense for syncs with an input state)
+        if command == "all":
+            tasks = []
+            for _command in COMMANDS:
+                tasks.extend([
+                    dispatch(
+                        connector.container,
+                        FileBackend(f"{output_directory}/{connector.version}/{_command}"),
+                        f"{output_directory}/{connector.version}",
+                        _command,
+                        config,
+                        catalog,
+                        state,
+                    ) for connector in [control_connector, target_connector]
+                ])
+        else:
+            tasks = [
+                dispatch(
+                    connector.container,
+                    FileBackend(f"{output_directory}/{connector.version}/{command}"),
+                    f"{output_directory}/{connector.version}",
+                    command,
+                    config,
+                    catalog,
+                    state,
+                ) for connector in [control_connector, target_connector]
+            ]
+        await asyncio.gather(*tasks)
+
+        await DiffComparator(client, output_directory).compare(control_connector, target_connector)
+
+
+async def dispatch(
+    container: dagger.Container,
+    backend: BaseBackend,
+    output_directory: str,
+    command: str,
+    config: Optional[SecretDict],
+    catalog: Optional[ConfiguredAirbyteCatalog],
+    state: Optional[Dict],
+):
+    runner = ConnectorRunner(container, backend, f"{output_directory}/{command}")
+
+    if command == "check":
+        await runner.call_check(config)
+
+    elif command == "discover":
+        await runner.call_discover(config)
+
+    elif command == "read":
+        runner = ConnectorRunner(container, backend, f"{output_directory}/read")
```

Comment:
these command specific runner can be removed right, the `runner` defined on line 92 can be reused.